src/postgres/commands/PostgresAccountWizard/PostgresServerNameStep.ts

```diff
@@ -43,11 +42,18 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
     }
 
     const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
+    const characterMatch = name.match(/^(?![-])[a-zA-Z0-9-]*(?<![-])$/);
+    const prefixSuffixMatch = name.match(/^[-]*.*[-]*$/);
     const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
 
     if (!availability.nameAvailable) {
-
-        return availability.message;
+        if (availability.reason === 'AlreadyExists') {
+            return localize('serverNameAvailabilityCheck', 'Server name "{0}" is not available.', name);
+        } else if (prefixSuffixMatch) {
+            return localize('serverNamePrefixSuffixCheck', 'Server name must not start or end in a hyphen.');
+        } else if (!characterMatch) {
```

Comment:
If the hyphen check is fixed, capital letters aren't caught with this check
<img width="509" alt="Screen Shot 2020-05-07 at 1 19 51 PM" src="https://user-images.githubusercontent.com/22795803/81340985-a5718b80-9065-11ea-85d1-050c1febf5b0.png">

We should be able to simplify the regex to this
```suggestion
        } else if (name.match(/[^a-z\d-]/)) {
```

Reply:
I like where Will is going, but a few nits:

```suggestion
        } else if (/[^a-z0-9-]/.test(name)) {
```


I prefer `<regexp>.test(<string>)` instead of `<string>.match(<regexp>)` because it returns a boolean instead of `RegExpMatchArray | null`. 
I also prefer `0-9` instead of `\d` because it feels more read-able to me. No need to remember that "d" stands for "digit", that it has to be `\` and not `/`, and that it has to be `d` and not `D` (which is the opposite)

Reply:
Thanks for catching, I missed that the uppercase letters were excluded for serverName